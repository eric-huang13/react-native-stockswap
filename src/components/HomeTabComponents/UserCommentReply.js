import React, {Component} from 'react';
import {
  Text,
  View,
  StyleSheet,
  Image,
  SafeAreaView,
  TouchableOpacity,
} from 'react-native';
import {moderateScale} from '../../util/responsiveFont';

export default class UserCommentReply extends Component {
  constructor(props) {
    super(props);
    this.state = {
      shouldShow: false,
    };
  }
  accountId = this.props.userAccount.id;

  navigationByCondition = (item) => {
    const {navigation} = this.props;
    if (item.userId === this.accountId) {
      navigation.navigate({
        name: 'MyProfile',
        params: {id: item.id},
      });
    } else {
      navigation.navigate({
        name: 'Profile',
        params: {id: item.userId},
      });
    }
  };
  render() {
    const {reply, id} = this.props;
    const {shouldShow} = this.state;

    const filteredReply = reply.filter((reply) => reply.commentId === id);

    return (
      <SafeAreaView style={style.mainContainer}>
        {this.state.shouldShow && filteredReply.length > 0 ? (
          <View>
            <TouchableOpacity
              onPress={() =>
                this.setState({
                  shouldShow: !shouldShow,
                })
              }>
              <Text style={style.replyButton}>Hide replies</Text>
            </TouchableOpacity>
            {filteredReply.map((item) => (
              <View key={item.id} style={style.itemContainer}>
                <View style={style.detailsContainer}>
                  <Image
                    style={style.postUserImage}
                    source={{uri: item.profileImg}}
                  />
                  <View style={style.nameBodyContainer}>
                    <TouchableOpacity
                      key={item.id}
                      onPress={() => this.navigationByCondition(item)}>
                      <Text style={style.name}>{item.name} </Text>
                    </TouchableOpacity>

                    <Text style={style.body}>{item.body} </Text>
                  </View>
                </View>
                <View style={style.belowCommentContainer}>
                  <Text style={style.time}>{item.time}</Text>
                  <View style={style.likesContainer}>
                    <Text style={style.likes}>{item.likes} likes</Text>
                    <Text style={style.reply}>Reply </Text>
                  </View>
                </View>
              </View>
            ))}
          </View>
        ) : !this.state.shouldShow && filteredReply.length === 1 ? (
          <TouchableOpacity
            onPress={() =>
              this.setState({
                shouldShow: !shouldShow,
              })
            }>
            <Text style={style.replyButton}>
              View {filteredReply.length} reply
            </Text>
          </TouchableOpacity>
        ) : !this.state.shouldShow && filteredReply.length > 0 ? (
          <TouchableOpacity
            onPress={() =>
              this.setState({
                shouldShow: !shouldShow,
              })
            }>
            <Text style={style.replyButton}>
              View {filteredReply.length} replies
            </Text>
          </TouchableOpacity>
        ) : null}
      </SafeAreaView>
    );
  }
}

const style = StyleSheet.create({
  mainContainer: {
    flexDirection: 'column',
    backgroundColor: '#2a334a',
    paddingVertical: moderateScale(4),
    paddingLeft: moderateScale(60),
    marginTop: moderateScale(10),
  },
  itemContainer: {
    marginBottom: moderateScale(16),
  },
  detailsContainer: {
    flexDirection: 'row',
    paddingRight: moderateScale(4),
  },
  nameBodyContainer: {
    flex: 1,
    flexDirection: 'column',
  },

  commentContainer: {
    flexDirection: 'row',
    paddingHorizontal: moderateScale(8),
    flexWrap: 'wrap',
    marginHorizontal: moderateScale(6),
    marginVertical: moderateScale(8),
  },

  name: {
    fontSize: 15,
    marginLeft: moderateScale(8),
    color: '#FFFFFF',
    fontFamily: 'Montserrat-Bold',
    marginBottom: moderateScale(3),
  },
  body: {
    marginLeft: moderateScale(8),
    color: 'lightgrey',
    fontSize: moderateScale(13.5),
    fontFamily: 'Montserrat-Medium',
  },
  postUserImage: {
    height: moderateScale(28),
    width: moderateScale(28),
    borderRadius: moderateScale(50),
  },
  belowCommentContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingLeft: moderateScale(36.5),
    paddingRight: moderateScale(6),
    alignItems: 'center',
    marginTop: moderateScale(4),
  },
  time: {
    fontSize: moderateScale(12.5),
    color: 'lightgrey',
    fontFamily: 'Montserrat-Italic',
  },
  likesContainer: {
    flexDirection: 'row',
  },
  likes: {
    color: '#FFFFFF',
    fontFamily: 'Montserrat-SemiBold',
    fontSize: moderateScale(13),
  },
  reply: {
    color: '#B8A0FF',
    marginLeft: moderateScale(16),
    fontFamily: 'Montserrat-SemiBold',
    fontSize: moderateScale(13),
  },
  replyButton: {
    color: '#B8A0FF',
    marginLeft: moderateScale(16),
    fontFamily: 'Montserrat-SemiBold',
    fontSize: moderateScale(13),
    marginBottom: moderateScale(10),
  },
});
