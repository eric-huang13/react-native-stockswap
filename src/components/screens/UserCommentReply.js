import React, {Component} from 'react';
import {
  Text,
  View,
  StyleSheet,
  Image,
  SafeAreaView,
  TouchableOpacity,
} from 'react-native';

export default class UserCommentReply extends Component {

  accountId = this.props.userAccount.id

  navigationByCondition = item => {
   const {navigation} = this.props;
   if (item.userId === this.accountId) {
     navigation.navigate({
       name: 'MyProfile',
       params: {id: item.id},
      })
   } else {
     navigation.navigate({
       name: 'Profile',
       params: {id: item.userId} 
     })
   }
 };
  render() {
    const {reply, id} = this.props;
    const filteredReply = reply.filter((reply) => reply.commentId === id);

    return (
      <SafeAreaView style={style.mainContainer}>
        {filteredReply.map((item) => (
          <View key={item.id} style={style.itemContainer}>
            <View style={style.detailsContainer}>
              <Image
                style={style.postUserImage}
                source={{uri: item.profileImg}}
              />
              <View style={style.nameBodyContainer}>
                <TouchableOpacity
                  key={item.id}
                  onPress={()=>this.navigationByCondition(item)             
                  }>
                  <Text style={style.name}>{item.name} </Text>
                </TouchableOpacity>

                <Text style={style.body}>{item.body} </Text>
              </View>
            </View>
            <View style={style.belowCommentContainer}>
              <Text style={style.time}>{item.time}</Text>
              <View style={style.likesContainer}>
                <Text style={style.likes}>{item.likes} likes</Text>
                <Text style={style.reply}>Reply </Text>
              </View>
            </View>
          </View>
        ))}
      </SafeAreaView>
    );
  }
}

const style = StyleSheet.create({
  mainContainer: {
    flexDirection: 'column',
    backgroundColor: '#2a334a',
    paddingVertical: 4,
    paddingLeft: 60,
    marginTop: 10,
  },
  itemContainer: {
    marginBottom: 16,
  },
  detailsContainer: {
    flexDirection: 'row',
    // marginBottom: 2,
    paddingRight: 4,
  },
  nameBodyContainer: {
    flex: 1,
    flexDirection: 'column',
  },

  commentContainer: {
    flexDirection: 'row',
    paddingHorizontal: 8,
    flexWrap: 'wrap',
    marginHorizontal: 6,
    marginVertical: 8,
  },

  name: {  
    marginLeft: 8,
    fontSize: 15,
    marginLeft: 8,
    color: '#FFFFFF',
    fontFamily:'Montserrat-Bold',
    marginBottom:3,
  },
  body: {    
    marginLeft: 8,
    color: 'lightgrey',
    fontSize: 13.5,
    fontFamily:'Montserrat-Medium',
  },
  postUserImage: {
    height: 28,
    width: 28,
    borderRadius: 50,
  },
  belowCommentContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingLeft: 36.5,
    paddingRight: 6,
    alignItems: 'center',
    marginTop: 4,
  },
  time: {
    fontSize: 12.5,
    color: 'lightgrey',
    fontFamily:'Montserrat-Italic',
  },
  likesContainer: {
    flexDirection: 'row',
  },
  likes: {
    color: '#FFFFFF',
    fontFamily:'Montserrat-SemiBold',
    fontSize:13,
  },
  reply: {
    color: '#B8A0FF',
    marginLeft: 16,
    fontFamily:'Montserrat-SemiBold',
    fontSize:13,
  },
});
