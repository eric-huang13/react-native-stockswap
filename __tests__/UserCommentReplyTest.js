import React from 'react';
import UserCommentReply from '../src/components/HomeTabComponents/UserCommentReply';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));
  

test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render   (<UserCommentReply userAccount={{id:'test' }} reply={[]}/>);
    const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});