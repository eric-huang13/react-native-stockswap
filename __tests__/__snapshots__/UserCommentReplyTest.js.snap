// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#2a334a",
      "flexDirection": "column",
      "marginTop": 14.124087591240876,
      "paddingLeft": 84.74452554744525,
      "paddingVertical": 5.649635036496351,
    }
  }
/>
`;
